containers/jml-extract-lambda/src/var/task/handler.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime as dt
+from urllib.error import HTTPError
+
+import awswrangler as wr
+import boto3
+import pandas as pd
+from notifications_python_client import prepare_upload
+from notifications_python_client.notifications import NotificationsAPIClient
+
+
+def handler(
+    event,
+    context,
+):
+    boto3.setup_default_session(region_name="eu-west-2")
+    secrets_client = boto3.client("secretsmanager")
+    response = secrets_client.get_secret_value(
+        SecretId="gov-uk-notify/production/api-key",
+    )
+    api_key = response["SecretString"]
+    notifications_client = NotificationsAPIClient(api_key)
+
+    now = dt.now()
+    current_date = dt.strftime(now.date(), "%d/%m/%Y")
+
+    query = f"""fields detail.data.user_id as user, detail.data.user_name as email
+    | filter detail.data.type = "s"
+    | filter detail.data.connection = "github"
+    | stats max(@timestamp) as last_login by "{current_date}" as effective_date, user, email
+    | sort last_login desc
+    """
+    year = dt.now().year
+    current_month = dt.now().month
+    previous_month = current_month - 1
+    end_datetime = dt(year, current_month, 1, 0, 0, 0)
+    if previous_month == 0:
+        previous_month = 12
+        year = year - 1
+        end_datetime = dt(year, current_month, 1, 0, 0, 0)
+
+    start_datetime = dt(year, previous_month, 1, 0, 0, 0)
+
+    dataframe = wr.cloudwatch.read_logs(
+        log_group_names=["/aws/events/auth0/alpha-analytics-moj"],
+        query=query,
+        start_time=start_datetime,
+        end_time=end_datetime,
+    )
+
+    dataframe["last_login"] = pd.to_datetime(
+        dataframe["last_login"], format="%Y-%m-%d %H:%M:%S.%f"
+    ).apply(lambda x: str(dt.strftime(x, "%Y/%m/%d")))
+
+    dataframe.to_excel("/tmp/test.xlsx", index=False)
```

Comment:
is `test.xlsx` the name of the attachment on the email?

Reply:
It will be (although needs replacing with a non-placeholder name - 
```suggestion
    datestamp = end_datetime.strftime(format="%Y_%m_%d")
    dataframe.to_excel("/tmp/jml_extract_.xlsx", index=False)
```